src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -54,6 +54,12 @@ public class Frame<TContext> : Frame
 
                     InitializeHeaders();
 
+                    if (_corruptedRequest)
+                    {
+                        await ProduceEnd();
+                        return;
+                    }
+
```

Comment:
This seems different than what we do for corrupted request headers. I would like to determine the correct behavior and consolidate this logic.